cdap-app-fabric/src/main/java/io/cdap/cdap/support/handlers/SupportBundleHttpHandler.java

```diff
@@ -65,10 +73,10 @@ public void createSupportBundle(HttpRequest request, HttpResponder responder,
                                   @Nullable @QueryParam("application") String application,
                                   @Nullable @QueryParam("programType") @DefaultValue("workflows") String programType,
                                   @Nullable @QueryParam("programId") @DefaultValue("DataPipelineWorkflow")
-                                      String programName,
-                                  @Nullable @QueryParam("run") String run,
+                                    String programName, @Nullable @QueryParam("run") String run,
                                   @Nullable @QueryParam("maxRunsPerProgram") @DefaultValue("1")
-                                      Integer maxRunsPerProgram) throws Exception {
+                                    Integer maxRunsPerProgram) throws Exception {
+    ensureVisibilityOnProgram(namespace, application, programType, programName, run);
```

Comment:
If all of these strings should not be null, can we check it before calling the visibility check? That way we can just throw an exception for each one without all of the nesting, like so:
```
if (namespace == null) {
  throw new IllegalArgumentException("Namespace must not be empty");
}
if (application == null) {
  throw new IllegalArgumentException("Application must not be empty");
}
...
```

Reply:
yeah because namespace and application can be nullable, that way we will grab all namespace from this instance